marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/entities/search/EntitySearchManagerTest.java

```diff
@@ -316,4 +312,22 @@ private void validateSearchWithTransform(String databaseType) {
                 "entityProperties so that the UI knows what structured values to show for each entity instance");
         assertTrue(node.get("results").get(1).has("entityProperties"));
     }
+
+    private void validateAllDataSearch(String databaseType) {
+        DatabaseClient client = databaseType.equalsIgnoreCase("staging") ? getHubClient().getStagingClient() : getHubClient().getFinalClient();
+        runAsDataHubDeveloper();
+        ReferenceModelProject project = installOnlyReferenceModelEntities(true);
+        deployEntityIndexes();
+        project.createCustomerInstance(new Customer(1, "Jane"), databaseType);
+        project.createCustomerInstance(new Customer(2, "Sally"), databaseType);
+
+        runAsHubCentralUser();
+
+        String query = "cts.estimate(cts.trueQuery())";
+        int count = Integer.parseInt(client.newServerEval().javascript(query).evalAs(String.class));
+        StringHandle results = new EntitySearchManager(getHubClient(), databaseType).search(new SearchQuery());
+        ObjectNode node = readJsonObject(results.get());
+        assertEquals(count, node.get("total").asInt(), "OOTB datahub artifacts, Customer and Order entity models and " +
```

Comment:
The assertion message shouldn't assume what's being returned. The benefit of doing an estimate on trueQuery is that you get a count of all the docs that an HC user can see. Then, you want to make sure that when you do an empty search, you get the same count back. You don't care what the count is, nor do you care what docs should be there - you just care that the two counts are the same. So the assertion message should be along the lines of:

"Expected " + count + " total documents; an empty search should result in a count equal to all the docs that the user can read in the database"